frontend/app/src/components/defi/LoanInfo.vue

```diff
@@ -0,0 +1,387 @@
+    <v-col>
+      <span class="font-weight-light subtitle-2">
+        Please select a loan to see information
+      </span>
+    </v-col>
+  </v-row>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue } from 'vue-property-decorator';
+import { createNamespacedHelpers } from 'vuex';
+import BaseExternalLink from '@/components/base/BaseExternalLink.vue';
+import WatcherDialog from '@/components/dialogs/WatcherDialog.vue';
+import AmountDisplay from '@/components/display/AmountDisplay.vue';
+import PremiumCard from '@/components/display/PremiumCard.vue';
+import StatCard from '@/components/display/StatCard.vue';
+import HashLink from '@/components/helper/HashLink.vue';
+import PremiumLock from '@/components/helper/PremiumLock.vue';
+import { WatcherArgs } from '@/services/types-api';
+import { MakerDAOVault, Watcher } from '@/services/types-model';
+import { MakerDAOVaultModel } from '@/store/balances/types';
+import { VaultEventsList } from '@/utils/premium';
+
+const { mapState, mapGetters } = createNamespacedHelpers('session');
+const { mapGetters: mapBalanceGetters } = createNamespacedHelpers('balances');
+
+@Component({
+  components: {
+    BaseExternalLink,
+    PremiumCard,
+    PremiumLock,
+    HashLink,
+    AmountDisplay,
+    StatCard,
+    VaultEventsList,
+    WatcherDialog
+  },
+  computed: {
+    ...mapState(['premium']),
+    ...mapGetters(['dateDisplayFormat']),
+    ...mapBalanceGetters(['loanWatchers'])
+  }
+})
+export default class LoanInfo extends Vue {
+  @Prop({ required: true })
+  vault!: MakerDAOVault | MakerDAOVaultModel | null;
+
+  premium!: boolean;
+  dateDisplayFormat!: string;
+  showWatcherDialog: boolean = false;
+  watcherMessage: string = '';
+  watcherVaultId: number | null = null;
+  loanWatchers!: Watcher[];
+
+  get watchers(): Watcher[] {
+    if (this.vault) {
+      return this.loanWatchers.filter(watcher => {
+        const watcherArgs = watcher.args as WatcherArgs['makervault_collateralization_ratio'];
+
+        // if (watcherArgs.indexOf(this.vault_id) >-1)
+        if (watcherArgs.vault_id.indexOf(String(this.vault!.identifier)) > -1)
+          return watcher;
+      });
+      // this.loanWatchers
+      //   .map(watcher => {
+      //     const watcherArgs = watcher.args as WatcherArgs['makervault_collateralization_ratio'];
+      //     return watcherArgs.vault_id;
+      //   })
+      //   .indexOf(String(this.vault.identifier)) > -1;
+    }
+    return [];
+  }
+
+  openWatcherDialog(vault: MakerDAOVault) {
+    this.showWatcherDialog = true;
+    this.watcherVaultId = vault.identifier;
+    this.watcherMessage = `Add / Edit / Delete watchers for Maker Vault #${vault.identifier} with current collateralization ratio ${vault.collateralizationRatio} and liquidation ratio ${vault.liquidationRatio}.`;
+  }
+
+  openLink(url: string) {
+    this.$interop.openUrl(url);
+  }
+
+  addWatcher() {
+    this.showWatcherDialog = false;
+    window.alert('api thing in WatcherDialog');
```

Comment:
Red alert, red alert :P

Reply:
Hah. I didn't catch this as it's never firing anymore.

Reply:
hahaahaha

damn I should start looking at the front-end code closely too now. I can imagine finding something like: `console.log("I wonder if lefteris even checks those")`